### PR TITLE
fix: toastError -> fix error messages (shows 'Something went wrong' i…

### PR DESCRIPTION
**Pull request by sofa-tata**
…nstead of meaningful message)

### CODE REVIEW

**Review by gonenidan — APPROVED** [review]

File: lib/utils.tsx [hunk 1]
```diff
@@ -104,19 +104,37 @@ const utils = {
...
         message = errorWithData.data
+      } else if (
+        errorWithData.data &&
+        typeof errorWithData.data === 'object' &&
+        'error' in errorWithData.data
+      ) {
+        message =
+          typeof errorWithData.data.error === 'string'
+            ? errorWithData.data.error
+            : 'Something went wrong'
```
Comment (udia76) [inline lib/utils.tsx:137]: Ww can simplify it using optional chaining, WDYT? 

` const message = 
    typeof err === 'string' ? err :
    err instanceof Error ? err.message :
    typeof err === 'object' && err?.data?.error ? err.data.error :
    'Something went wrong';`

Comment (sofa-tata) [reply lib/utils.tsx:137]: I think in this case it's more readable with ifs

Comment (udia76) [reply lib/utils.tsx:137]: If you prefer if statements, let's simplify the cases ? 

`let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err instanceof Error) {
  message = err.message;
} else if (typeof err === 'object' && err?.data?.error) {
  message = err.data.error;
} else {
  message = 'Something went wrong';
}`

Comment (sofa-tata) [reply lib/utils.tsx:137]:
```
else if (err instanceof Error) {
message = err.message;
}
```
⬆️ there are cases when `err` object is not `instanceof Error` and still can have the `message` property, therefore I changed this condition

```
else if (typeof err === 'object' && err?.data?.error) {
message = err.data.error;
}
```

⬆️ we also need to check that `err.data.error` is a string before assigning it to message

Comment (udia76) [reply lib/utils.tsx:137]: ok , so we can try something like:

```
let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err && typeof err === 'object' && 'message' in err && typeof err.message === 'string') {
  message = err.message;
} else if (err && typeof err === 'object' && err.data && typeof err.data === 'object' && typeof err.data.error === 'string') {
  message = err.data.error;
} else {
  message = 'Something went wrong';
}
```

Comment (sofa-tata) [reply lib/utils.tsx:137]: this way we're not covering this case:
```
if (typeof err.data === 'string') {
        message = err.data // errorWithData.data in the code
      }
```

Comment (udia76) [reply lib/utils.tsx:137]: ok, so let's try that one:

```
let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err && typeof err === 'object' && 'message' in err && typeof err.message === 'string') {
  message = err.message;
} else if (err && typeof err === 'object' && err.data) {
  if (typeof err.data === 'string') {
    message = err.data;
  } else if (typeof err.data === 'object' && 'error' in err.data && typeof err.data.error === 'string') {
    message = err.data.error;
  } else {
    message = 'Something went wrong';
  }
} else {
  message = 'Something went wrong';
}
```

**Review by udia76 — APPROVED** [review]

**Pull request merged**